server/api-util/lineItemHelpers.js

```diff
@@ -38,15 +38,15 @@ exports.calculateTotalPriceFromQuantity = (unitPrice, unitCount) => {
  * @returns {Money} lineTotal
  */
 exports.calculateTotalPriceFromPercentage = (unitPrice, percentage) => {
-  const numericPrice = convertMoneyToNumber(unitPrice);
-  const numericTotalPrice = new Decimal(numericPrice)
+  const amountFromUnitPrice = getAmountAsDecimalJS(unitPrice);
+  const totalPrice = amountFromUnitPrice
     .times(percentage)
     .dividedBy(100)
-    .toNumber();
-  return new Money(
-    convertUnitToSubUnit(numericTotalPrice, unitDivisor(unitPrice.currency)),
-    unitPrice.currency
-  );
+    .round();
```

Comment:
I can see that the default rounding mode in decimal.js is half up (https://mikemcl.github.io/decimal.js/#rounding) but I'm wondering if that's a stable default we should just trust to be there. It would introduce a hard to spot bug if some future version would change the default mode. That might be unlikely but explicitly setting the rounding mode would eliminate that possibility. Another reason to be explicit is to document it clearly that that's what we're doing. Maybe that could be set in `getAmountAsDecimalJS`?

Reply:
@ovan I'm not sure if that is something we should be afraid of - the whole math library commits suicide if they change their default rounding mode. 

Anyway, `round()` doesn't take rounding mode as an optional parameter. It uses a library-wide variable for that - and I don't want to hard-code that variable in a random file. 
So, I'll change this to `.toNearest(1, Decimal.ROUND_HALF_UP)`.

Reply:
I added these now here. `getAmountAsDecimalJS` is called before the calculations, so it's not the right place. And to be frank, I like to remind people that there are constraints that they need to be aware of.

So, the code now looks like this:
```js
  // NOTE: We round the amount to the nearest integer.
  //       Payment processors don't support fractional subunits.
  const totalPrice = amountFromUnitPrice
    .times(percentage)
    .dividedBy(100)
    .toNearest(1, Decimal.ROUND_HALF_UP);
```

Reply:
@Gnito I think that's perfect. I didn't realize `toNeareast` exists and was only looking at `round` which didn't take rounding mode as a parameter. Great that this one does and the rounding mode is now clearly visible here. 👍